backend/src/services/secret-replication/secret-replication-service.ts

```diff
@@ -0,0 +1,342 @@
+                      secretValueIV: doc.secretValueIV,
+                      secretValueTag: doc.secretValueTag,
+                      secretValueCiphertext: doc.secretValueCiphertext,
+                      secretBlindIndex: doc.secretBlindIndex,
+                      secretCommentIV: doc.secretCommentIV,
+                      secretCommentTag: doc.secretCommentTag,
+                      secretCommentCiphertext: doc.secretCommentCiphertext,
+                      isReplicated: true,
+                      skipMultilineEncoding: doc.skipMultilineEncoding
+                    }
+                  };
+                })
+              });
+              nestedImportSecrets = nestedImportSecrets.concat(
+                newSecrets.map(({ id, version }) => ({ operation: SecretOperations.Update, version, id }))
+              );
+            }
+            if (locallyDeletedSecrets.length) {
+              const newSecrets = await secretDAL.delete(
+                {
+                  $in: {
+                    id: locallyDeletedSecrets.map(({ id }) => id)
+                  },
+                  isReplicated: true,
+                  folderId: importFolderId
+                },
+                tx
+              );
+              nestedImportSecrets = nestedImportSecrets.concat(
+                newSecrets.map(({ id, version }) => ({ operation: SecretOperations.Delete, version, id }))
+              );
+            }
+          });
+
+          await queueService.queue(QueueName.SecretReplication, QueueJobs.SecretReplication, {
+            folderId: importedFolder.id,
+            projectId,
+            secrets: nestedImportSecrets,
+            secretPath: importedFolder.path,
+            environmentId: importedFolder.envId,
+            actorId,
+            actor
+          });
+          const folderLock = await keyStore
+            .acquireLock([`secret-replication-${importFolderId}`], 5000)
+            .catch(() => null);
+          if (folderLock) {
+            await snapshotService.performSnapshot(importFolderId);
+            await folderLock.release();
+            await secretQueueService.syncSecrets({
+              excludeReplication: true,
+              projectId,
+              secretPath: importedFolder.path,
+              environmentSlug: importedFolder.environmentSlug
+            });
+          }
+        }
+
+        await keyStore.setItemWithExpiry(
+          keystoreReplicationSuccessKey(job.id as string, secretImport.id),
+          SECRET_IMPORT_SUCCESS_LOCK,
+          1,
+          KeyStorePrefixes.SecretReplication
+        );
+      }
+      await secretVersionDAL.update({ $in: { id: replicatedSecrets.map(({ id }) => id) } }, { isReplicated: true });
+      /*  eslint-enable no-await-in-loop */
+    } finally {
+      await lock.release();
+    }
+  });
+
+  queueService.listen(QueueName.SecretReplication, "failed", (job, err) => {
+    logger.error(err, "Failed to replicate secret", job?.data);
+  });
```

Comment:
for failed replication jobs, how do we plan on giving feedback to the users?

Reply:
We don't. At present we still don't have a proper place to show case failed replication.

Reply:
do you think we could somehow add an error icon beside the "resync replicated secrets"? 
<img width="644" alt="image" src="https://github.com/Infisical/infisical/assets/65645666/d600a2d5-f2bb-4710-89dd-3e2d48dd570a">

Reply:
I did think about this but also i think it had some issues with it too. Let me look into it again

Reply:
Added a feedback messages